background.js

```diff
@@ -16,47 +21,146 @@ setInterval(() => {
+        // })
+        // .catch(error => {
+        //   console.log('send failure');
+        // });
+        chrome.tabs.executeScript(
+          tab.id,
+          {
+            file: 'get-spotify-info.js',
+          },
+          scrapedSong => {
+            console.log('scrapedSong', scrapedSong);
+            if (scrapedSong[0]) {
+                body.songName = scrapedSong[0].songName;
+                body.albumURL = scrapedSong[0].albumURL;
+                body.artist = scrapedSong[0].artistString;
+                body.artistArray = scrapedSong[0].artistArray
+                body.currentTime = scrapedSong[0].currentTime;
+                body.durationTime = scrapedSong[0].durationTime;
+                body.origin = scrapedSong[0].origin;
+                body.domain = scrapedSong[0].domain;
+                body.album = scrapedSong[0].albumName; //* potentially empty string
+                body.albumDate = scrapedSong[0].playlistPublishDate; //* potentially empty string
+                body.albumTotal = scrapedSong[0].playlistTotal; //* potentially empty string
+                body.songNumber = scrapedSong[0].songNumber; //* potentially empty string
+    
+                postData('http://localhost:4242/api/song', body);
+            }
+          }
+        );
+      }
+      if(/youtube\.com/.test(tab.url)) {
+        console.log(tab.title);
+        tab.title.split(' - ').map((chunk, index, chunks) => {
+          if (index === 0) {
+            body.songName = chunk;
+          } else if (index === chunks.length - 1) {
+            body.origin = chunk;
+          }
+        });
+        // postData('http://localhost:4242/api/album', body)
+        // .then(result => {
+        //   console.log('send success');
+        // })
+        // .catch(error => {
+        //   console.log('send failure');
+        // });
+        var config = {"startingTimer":startingTimer,"currentSongURL":tab.url,"currentSongTime":currentSongTime,"currentSong":currentSong};
+        chrome.tabs.executeScript(tab.id, {
+          code: 'var config = ' + JSON.stringify(config)
+        }, 
+        function() {
+          chrome.tabs.executeScript(tab.id, {file: 'get-youtube-info.js'}, 
+          scrapedSong => {
+            console.log('scrapedSong', scrapedSong);
+            if (scrapedSong[0]) {
+              body.songName = scrapedSong[0].currentSongApprox!==undefined?scrapedSong[0].currentSongApprox:scrapedSong[0].songName; //USE APPROX UNLESS EMPTY
+              body.views = scrapedSong[0].views;
+              body.videoType = scrapedSong[0].type;
+              body.durationTime = scrapedSong[0].durationTime;
+              body.currentTimeSeconds = scrapedSong[0].currentTimeSeconds;
+              body.origin = scrapedSong[0].origin;
+              body.domain = scrapedSong[0].domain;
+              if(currentSong == scrapedSong[0].songName && currentSongTime == scrapedSong[0].currentTime){
+              
```

Comment:
A comment here explaining why we do nothing would be cool.

Couldn't the `if` logic just be inverted though?